File: 1468-check-if-n-and-its-double-exist/check-if-n-and-its-double-exist.cpp
class Solution {
public:
    bool checkIfExist(vector<int>& arr) {
        unordered_set<int> d;
        for(int i : arr){
            if(d.find(i*2)!=d.end()) return true;
            if(i%2==0 && d.find(i/2)!=d.end()) return true;
            d.insert(i);
        }
        return false;
    }
};

File: 1468-check-if-n-and-its-double-exist/check-if-n-and-its-double-exist.java
class Solution {
    public boolean checkIfExist(int[] arr) {
        HashSet<Integer> d = new HashSet<>();
        for(int i : arr){
            if(d.contains(i*2)) return true;
            if(i%2==0 && d.contains(i/2)) return true;
            d.add(i);
        }       
        return false;
    }
}